sound/soc/sof/mediatek/mt8195/mt8195.c

```
@@ -311,6 +311,41 @@ static int mt8195_dsp_remove(struct snd_sof_dev *sdev)
 	return 0;
 }
 
+static int mt8195_dsp_suspend(struct snd_sof_dev *sdev, u32 target_state)
+{
+	struct platform_device *pdev = container_of(sdev->dev, struct platform_device, dev);
+
+	/*stall and reset dsp*/
+	sof_hifixdsp_shutdown(sdev);
+
+	/*power down adsp sram*/
+	adsp_sram_power_on(&pdev->dev, false);
+
+	/*turn off adsp clock*/
+	adsp_clock_off(sdev);
```

maybe you don't want to return an error and prevent the system, but IF there's an error for clock_off and power_off it's worth logging it, at least as dev_warn(), no?

---

I will add return check for power off function, but for clock off function ,it only call clk_disable_unprepare without return value, so I won't add checking for it. Thanks.